processor/src/it/java/com/linecorp/decaton/processor/MetricsTest.java

```diff
@@ -70,4 +75,39 @@ public void testMetricsCleanup() throws Exception {
         List<Meter> meters = Metrics.registry().getMeters();
         assertEquals(emptyList(), meters);
     }
+
+    @Test(timeout = 30000)
+    public void testDetectStuckPartitions() throws Exception {
+        // Micrometer doesn't track values without at least one register implementation added
+        Metrics.register(new PrometheusMeterRegistry(PrometheusConfig.DEFAULT));
+
+        CountDownLatch processLatch = new CountDownLatch(PARTITIONS * 2);
```

Comment:
just `new CountDownLatch(PARTITIONS)` not work ?

Reply:
so after very tough try & fails finally I've managed to remove all those flacky-ish stuff from this test, along with fixing core to regard `decaton.max.pending.records` at exact boundary: cbc7d75 

it could still be failed on race scenario like this test confirms paused count before subscription thread to reach `pausePartitions` but its possibility is likely very low. we cannot await on next records arrival either because all partitions are already paused and processors won't get any of them.